### PR TITLE
feat(props): expose server properties

### LINKED ISSUE #230: Add APIs to expose server-owned properties
It would be nice to have some APIs that expose the server-owned properties kept by the device.

examples:
1. `get_server_owned_property(interface, path)`
2. `get_all_server_owned_properties()`

### PR DESCRIPTION
**Pull request by joshuachp**
Add a field for the ownership to the properties and trait to access them form the AstarteSDK.

Closes #230

### CODE REVIEW

**Review by harlem88 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,17 +8,25 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Make the `DynError` trait bound shareable across threads.
+- Added ownership field to the `StoredProp` struct.
+- The `PropertyStore::store_prop` now receives the `StoredProp` struct.
 
 ### Fixed
 - Unset of property send empty buffer instead of document with null value.
 
+### Deprecated
+- Warn the `AstarteDeviceSdk::get_property` method to use the `PropAccess`
+  trait instead
```
Comment (sorru94) [inline CHANGELOG.md:28]:
```suggestion
- The `AstarteDeviceSdk::get_property` method. Use the `PropAccess`  trait 
instead.
```

Comment (joshuachp) [reply CHANGELOG.md]: Refrased

**Review by sorru94 — APPROVED** [review]

File: src/properties.rs [hunk 2]
```diff
@@ -71,12 +74,46 @@ where
 
         self.try_load_prop(&mapping, &path).await
     }
+
+    async fn interface_props(&self, interface: &str) -> Result<Vec<StoredProp>, Error> {
+        let interfaces = &self.interfaces.read().await;
+        let prop = interfaces
```
Comment (harlem88) [inline src/properties.rs:80]: What do you think about renaming the parameter of interface_props from `interface` to `interface_name`? I would like renaming also the `let prop` to `let interface`, I think `prop` is misleading.

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/properties.rs (hunk 2 above)
Comment (joshuachp) [reply src/properties.rs:80]: I agree on the `interface_name`, but for the prop the type is `PropertyRef` which I think is better represented by prop.

**Review by harlem88 — APPROVED** [review]

**Review by sorru94 — APPROVED** [review]

**Review by rgallor — APPROVED** [review]

**Review by rgallor — APPROVED** [review]
I've left a suggestion which is not related to the implemented functionalities.

**Review by rgallor — APPROVED** [review]
I've left 2 suggestions which are not related to the implemented functionalities.

**Pull request merged**